main.go

```diff
@@ -1,21 +1,64 @@
 package main
 
 import (
+	"context"
 	"flag"
+	"log"
 
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/internal/provider"
+	oldprovider "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server"
+	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
+	"github.com/hashicorp/terraform-plugin-mux/tf6muxserver"
 )
 
-const ProviderAddr = "registry.terraform.io/Snowflake-Labs/snowflake"
+var version string = "dev" // goreleaser can pass other information to the main package, such as the specific commit
+// https://goreleaser.com/cookbooks/using-main.version/
 
 func main() {
```

Comment:
Let's test this even manually, so that we are sure, that we are not introducing any incompatibility with the setup change (because as we discussed some things are slipping through our tests)

Reply:
Closing after verbal discussion